Documentation/Scripts/fetchRefs.sh

```diff
@@ -6,43 +6,49 @@ ALLBOOKS="HTTP AQL Manual Cookbook Drivers"
 
 GITAUTH="$1"
 
-for book in ${ALLBOOKS}; do 
+for book in ${ALLBOOKS}; do
 
     repos=$(grep '^ *<!-- SYNC: ' "../Books/${book}/SUMMARY.md" |sed -r 's;^ *<!-- SYNC: (.+) -->$;\1;')
+    reposUnique=$(awk '!seen[$0]++' <(for oneRepo in ${repos}; do echo "${oneRepo}"; done))
 
-    for oneRepo in ${repos}; do
+    for oneRepo in ${reposUnique}; do
 
         REPO=$(echo     "$oneRepo" |cut -d ';' -f 1)
         CLONEDIR=$(echo "$oneRepo" |cut -d ';' -f 2)
         SUBDIR=$(echo   "$oneRepo" |cut -d ';' -f 3)
         SRC=$(echo      "$oneRepo" |cut -d ';' -f 4)
         DST=$(echo      "$oneRepo" |cut -d ';' -f 5)
 
-
         CODIR="../Books/repos/${CLONEDIR}"
         AUTHREPO="${REPO/@/${GITAUTH}@}"
-        if test -d "${CODIR}"; then
-            (
-                cd "${CODIR}"
-                git pull --all
-            )
-        else
-            if test "${GITAUTH}" == "git"; then
-                AUTHREPO=$(echo "${AUTHREPO}" | sed -e "s;github.com/;github.com:;" -e "s;https://;;" )
-            fi
-            git clone "${AUTHREPO}" "${CODIR}"
+
+        if test "${GITAUTH}" == "git"; then
+            AUTHREPO=$(echo "${AUTHREPO}" | sed -e "s;github.com/;github.com:;" -e "s;https://;;" )
         fi
 
+        echo
+        echo "Syncing ${AUTHREPO} -> ${CLONEDIR}"
+
         # extract branch/tag/... for checkout from VERSIONS file
         branch=$(grep "EXTERNAL_DOC_${CLONEDIR}=" "../../VERSIONS" | sed "s/^EXTERNAL_DOC_${CLONEDIR}=//")
 
         if [ -z "${branch}" ]; then
-            echo "no branch for ${CLONEDIR}, specify in VERSIONS file."
+            echo "Branch for ${CLONEDIR} needs to be specified in VERSIONS file!"
             exit 1
         fi
 
-        # checkout name from VERSIONS file and pull=merge origin
-        (cd "${CODIR}" && git checkout "${branch}" && git pull)
+        if test -d "${CODIR}"; then
+            # checkout branch as specified in VERSIONS file, update by pulling (ff or merge)
+            (
```

Comment:
error handling?

Reply:
There was no error handling before, or am I missing something?

Should it be `cd "${CODIR}" && git fetch && git checkout "${branch}" && git merge FETCH_HEAD` so that subsequent commands are not executed in case of an early error?